### DISCUSSION TITLE
HIP-11: Layer-2 Smart Contract Network

### DISCUSSION THREAD

**Discussion by lucashenning** [hiero-ledger/hiero-improvement-proposals#49] [Edits]
Opening this thread to discuss https://github.com/hashgraph/hedera-improvement-proposal/pull/42 

This HIP defines and discusses a way to create a network consisting of a plurality of nodes that provide an Ethereum-like smart contract service over Hedera at low cost and significantly higher transaction rate than the existing on-ledger Smart Contract service on Hedera.

**Comment by kenthejr** [comment]
Merged as https://github.com/hashgraph/hedera-improvement-proposal/blob/master/HIP/hip-11.md

**Comment by topshef** [comment]
I'm interested in the subject of this HIP, will dig in some more when I have a chance, thanks for sharing :-)

**Comment by parabhairava** [comment]
Another thing to look at is https://www.notoros.io/
Notoros does not only fully support EVM (they are integrating with Ethereum with the ability to fork the entire application layer), but they are very generalized so that they support other frameworks and environments. They are also very appreciative of Hedera and would be happy to integrate.

**Comment by rogeriob2br** [comment]
I was reading HIP-11, which deals with Layer 2 of Smart Contracts, and I would like to discuss whether we could include WASM support, similar to eWASM.

**Comment by sergmetelin** [comment]
Were all issues related to this HIP addressed? Can I move it to `Accepted` status? @lucashenning

**Comment by possible-panther** [comment]
I'm not sure I saw a proposal in the HIP for how payments get to the layer 2 network. Would it be valuable to have a separate HIP that allowed for custom fees to write to an HCS topic in the same way we have custom fees on tokens? I would like to see a situation where anyone can write to a topic but they must pay some specified account a minimum amount to do so.
